Sources/web3swift/Operations/WriteOperation.swift

```diff
@@ -11,8 +11,10 @@ import Core
 public class WriteOperation: ReadOperation {
 
     // FIXME: Rewrite this to CodableTransaction
-    public func writeToChain(password: String) async throws -> TransactionSendingResult {
-        await transaction.resolve(provider: web3.provider)
+    public func writeToChain(password: String, policies: Policies = .auto) async throws -> TransactionSendingResult {
+        // TODO: might change to be dependency
+        let resolver = PolicyResolver(provider: web3.provider)
+        try await resolver.resolveAll(for: &transaction)
```

Comment:
Though this one is what I like, it could be enhanced a bit with making this method with optional possibility to allow user change policies for exact that call (which is doing in current one implementation) but against it's default policies settings. not some predefined default value. I see it that way:

```swift
public func writeToChain(password: String, policies: Policies = web3.provider.policies) async throws -> TransactionSendingResult {
```
Just mention that we would made here not any given policy value, but will inherit from users general settings if any, and those one in turn could be set to `.auto` by default.